Packs/ExabeamDataLake/Integrations/ExabeamDataLake/ExabeamDataLake.py

```diff
@@ -0,0 +1,273 @@
+    """
+     Generate a list of dates within a specified range.
+
+    Args:
+        start_time: The start date of the range in the format "YYYY-MM-DD".
+        end_time: The end date of the range in the format "YYYY-MM-DD".
+
+    Raises:
+        DemistoException: If the start time is not before the end time, or if the difference between start time and end time is
+        greater than 10 days.
+
+    Returns:
+        list: A list of dates within the specified range, formatted as strings in the format "YYYY.MM.DD".
+    """
+    start_time = datetime.strptime(start_time, "%Y-%m-%d")
+    end_time = datetime.strptime(end_time, "%Y-%m-%d")
+
+    if start_time >= end_time:
+        raise DemistoException("Start time must be before end time")
+
+    if (end_time - start_time).days > 10:
+        raise DemistoException("Difference between start time and end time must be less than or equal to 10 days")
+
+    dates = []
+    current_date = start_time
+    while current_date <= end_time:
+        dates.append(current_date.strftime("%Y.%m.%d"))
+        current_date += timedelta(days=1)
+
+    return dates
+
+
+def get_date(time: str):
+    """
+    Get the date from a given time string.
+
+    Args:
+        time (str): The time string to extract the date from.
+
+    Returns:
+        str: The date extracted from the time string formatted in ISO 8601 format (YYYY-MM-DD),
+        or None if the time string is invalid.
+    """
+    date_time = arg_to_datetime(arg=time, arg_name="Start time", required=True)
+    if date_time:
+        date = date_time.strftime(ISO_8601_FORMAT)
+    return date
+
+
+def calculate_page_parameters(args: dict) -> tuple[int, int]:
+    """
+      Calculate the page parameters for pagination.
+
+    Args:
+        args: A dictionary containing the arguments passed to the function.
+
+    Raises:
+        DemistoException: If invalid combinations of arguments are provided. You can only provide 'limit'
+        alone or 'page' and 'page_size' together.
+
+    Returns:
+        tuple: A tuple containing two integers representing the 'from' and 'size' parameters for pagination.
+        'from' is the index of the first item to retrieve, and 'size' is the number of items to retrieve.
+    """
+    page_arg = args.get('page')
+    page_size_arg = args.get('page_size')
+    limit_arg = args.get('limit')
+
+    if (limit_arg and (page_arg or page_size_arg)) or ((not (page_arg and page_size_arg)) and (page_arg or page_size_arg)):
+        raise DemistoException("You can only provide 'limit' alone or 'page' and 'page_size' together.")
+
+    if args.get('page') and args.get('page_size'):
+        page = arg_to_number(args.get('page', '1'))
+        page_size = arg_to_number(args.get('page_size', '50'))
+        if page and page_size:
+            from_param = page * page_size - page_size
+            size_param = page_size
+    else:
+        from_param = 0
+        size_param = arg_to_number(args.get('limit', '50')) or 50
```

Comment:
The arg_to_number is a bit problematic since it returns Any | int 
```suggestion
        size_param = int(args.get('limit', '50'))
```